src/server/pkg/deploy/assets/assets.go

```diff
@@ -138,6 +138,18 @@ type StorageOpts struct {
 	UploadConcurrencyLimit int
 }
 
+const (
+	// RequireCriticalServersOnlyEnvVar is the environment variable for requiring critical servers only.
+	RequireCriticalServersOnlyEnvVar = "REQUIRE_CRITICAL_SERVERS_ONLY"
+)
+
+const (
+	// DefaultRequireCriticalServersOnly is the default for requiring critical servers only.
+	// (bryce) this default is set here and in the service env config, need to figure out how to refactor
+	// this to be in one place.
+	DefaultRequireCriticalServersOnly = false
```

Comment:
I could see arguments both ways, but I'm curious: why did we decide to default this to false?

Reply:
I made the default false because it made sense to me to fail fast if any server failed to come up. If the user is okay with the functionality of a noncritical server not being available, then they can set this flag. It basically ensures that a user is aware of a problem earlier rather than later. I can see the benefit of this defaulting to true for making starting out with Pachyderm easier though.